document/CoreFormat.mdk

```diff
@@ -775,26 +813,27 @@ A `result` object describes a single result produced by an analysis tool.
 
 ### `ruleId` property { #result-ruleId }
 
-A `result` object should contain a property named `ruleId` whose value is a string containing
+If the `level` property (&sect;[#result-level]) of a `result` object is anything other than `note`,
+then the `result` object shall contain a property named `ruleId` whose value is a string containing
 the stable, opaque identifier for the rule that was evaluated to produce the result.
 
 ~ IsoExample
 `"CA2101"`
 ~
 
-Analysis tools that produce the SARIF format shall provide a `ruleId` for each `result`
-(with the possible exception of `result`s whose `kind` property is `note`; see (&sect;[#result-kind]).
+If the `level` property of the `result` object is `note`,
+then the `ruleId` property may be absent.
```

Comment:
Not a typo. It renders `result` in the code font and then adds the "s":

`result`s